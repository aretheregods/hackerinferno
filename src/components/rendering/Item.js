import { Link } from 'inferno-router';

export function Item(props) {

    const item = props.item


    return (
        <div key={item.id} className="list-item">
            <div className="item-number">{(props.index + 1) + ((props.page_number - 1) * 30)}</div>
            <div className="detail-box">
                {item.type === 'ask' || item.type === 'show' ?
                <Link to={'/item/' + item.id} className="detail-child story-link"> {item.title} </Link> :
                <a href={item.url} className="detail-child story-link"> {item.title} </a>
                }
                {item.user &&
                    <Link to={'/user/' + item.user} className="detail-child user-link">by <span className="user-id">{item.user}</span></Link>
                }
                <p className="detail-child detail-box-bottom">
                {item.points || "0"}
                <span className="score-star">{String.fromCharCode(9734)}&nbsp;</span>
                {item.time_ago}&nbsp;
                <Link to={'/item/'  + item.id} className="bottom-child item-comments"> {!item.comments_count ?
                    <span className="item-comments-span">Discuss</span> :
                    <span className="item-comments-span">{item.comments_count} comments</span>
                }</Link>
                </p>
            </div>
        </div>
    )
    
}
