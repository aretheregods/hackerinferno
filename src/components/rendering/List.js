import { Link } from 'inferno-router';
import { Item } from './Item'

function Pager(props) {
    const endpoint_path = props.list;
    const page_number = props.page;
    const more_pages = props.more;
    const forward_path = !page_number ? 2 : page_number + 1;
    return (
        <span id="pager-span" style={{display:"flex","justify-content":"space-around"}}>
            {page_number > 1 ?
                <Link to={`/${endpoint_path}${page_number - 1}`}> {String.fromCharCode(8592)}Back</Link> :
                <a href={`/${endpoint_path}${page_number}`} style={{"pointer-events":"none",cursor:"default",color:"#D3D3D3"}}>{String.fromCharCode(8592)}Back</a>
            }
            {page_number}
            {more_pages ?
                <Link to={`/${endpoint_path}${forward_path}`}>More {String.fromCharCode(8594)}</Link> :
                <a href={`/${endpoint_path}${page_number}`} style={{"pointer-events":"none",cursor:"default",color:"#D3D3D3"}}>More {String.fromCharCode(8594)}</a>
            }
        </span>
    )
}

// The Pure Functional Component Format that all main list pages follow
export function HackerList(props) {

    // Data received from props
    const data = props.component_data
    const page_endpoint = props.component
    const page_number = props.page ? ~~props.page : 1;
    const endpoint_path = page_endpoint !==  'news' ? page_endpoint + '/' : '';


    // Basiic List Item Template
    const list_items = data.map(function(listobject, index) {
        return <Item item={listobject} index={index} page_number={page_number}/>
    });

    //  The List of paginated Stories
    return (
        <div id="hacker-list">
            <div id="items">{list_items}</div>
            <div id="pager">
                <h4>
                    <Pager list={endpoint_path} page={page_number} more={data.length === 30} />
                    {/* <span id="pager-span" style={{display:"flex","justify-content":"space-around"}}>
                        {page_number > 1 ?
                            <Link to={`/${endpoint_path}${page_number - 1}`}> {String.fromCharCode(8592)}Back</Link> :
                            <a href={`/${endpoint_path}${page_number}`} style={{"pointer-events":"none",cursor:"default",color:"#D3D3D3"}}>{String.fromCharCode(8592)}Back</a>
                        }
                        {page_number}
                        {more_pages ?
                            <Link to={`/${endpoint_path}${forward_path}`}>More {String.fromCharCode(8594)}</Link> :
                            <a href={`/${endpoint_path}${page_number}`} style={{"pointer-events":"none",cursor:"default",color:"#D3D3D3"}}>More {String.fromCharCode(8594)}</a>
                        }
                    </span> */}
                </h4>
            </div>
        </div>
    )
}
