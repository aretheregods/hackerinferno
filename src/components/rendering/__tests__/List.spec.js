import { render } from 'inferno';
import { renderToSnapshot } from 'inferno-test-utils';
import { HackerList, Pager } from '../List.js';
import { Item } from '../Item';

const some_items = [{
    id: 1,
    user: "a",
    title: "something",
    url: "https://whateverblah.com/123456",
    points: 1000,
    time_ago: "8 hours ago",
    comments_count: 200,
    type: "news"
},{
    id: 2,
    user: "b",
    title: "somethingelse",
    url: "https://whateverfuck.com/654321",
    points: 500,
    time_ago: "4 hours ago",
    comments_count: 100,
    type: "ask",
},{
    id: 3,
    user: "c",
    title: "somethingelseagain",
    url: "https://whateverblahfuck.com/1234567",
    points: 250,
    time_ago: "2 hours ago",
    comments_count: 50,
    type: "news"
}];

it('should render info about 3 stories in a div with a greyed out pager', () => {
    const snapshot = renderToSnapshot(<HackerList component_data={some_items} component="news" page={1}/>);
    expect(snapshot).toMatchSnapshot();
})
