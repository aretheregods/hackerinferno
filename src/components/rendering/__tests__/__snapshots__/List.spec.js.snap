// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render info about 3 stories in a div with a greyed out pager 1`] = `
<div
  id="hacker-list"
>
  <div
    id="items"
  >
    <div
      className="list-item"
    >
      <div
        className="item-number"
      />
      <div
        className="detail-box"
      >
        <a
          className="detail-child story-link"
          href="https://whateverblah.com/123456"
        >
           
          something
           
        </a>
        <a
          className="detail-child user-link"
          href="/user/a"
        >
          by 
          <span
            className="user-id"
          >
            a
          </span>
        </a>
        <p
          className="detail-child detail-box-bottom"
        >
          <span
            className="score-star"
          >
            ★
             
          </span>
          8 hours ago
           
          <a
            className="bottom-child item-comments"
            href="/item/1"
          >
             
            <span
              className="item-comments-span"
            >
               comments
            </span>
          </a>
        </p>
      </div>
    </div>
    <div
      className="list-item"
    >
      <div
        className="item-number"
      />
      <div
        className="detail-box"
      >
        <a
          className="detail-child story-link"
          href="/item/2"
        >
           
          somethingelse
           
        </a>
        <a
          className="detail-child user-link"
          href="/user/b"
        >
          by 
          <span
            className="user-id"
          >
            b
          </span>
        </a>
        <p
          className="detail-child detail-box-bottom"
        >
          <span
            className="score-star"
          >
            ★
             
          </span>
          4 hours ago
           
          <a
            className="bottom-child item-comments"
            href="/item/2"
          >
             
            <span
              className="item-comments-span"
            >
               comments
            </span>
          </a>
        </p>
      </div>
    </div>
    <div
      className="list-item"
    >
      <div
        className="item-number"
      />
      <div
        className="detail-box"
      >
        <a
          className="detail-child story-link"
          href="https://whateverblahfuck.com/1234567"
        >
           
          somethingelseagain
           
        </a>
        <a
          className="detail-child user-link"
          href="/user/c"
        >
          by 
          <span
            className="user-id"
          >
            c
          </span>
        </a>
        <p
          className="detail-child detail-box-bottom"
        >
          <span
            className="score-star"
          >
            ★
             
          </span>
          2 hours ago
           
          <a
            className="bottom-child item-comments"
            href="/item/3"
          >
             
            <span
              className="item-comments-span"
            >
               comments
            </span>
          </a>
        </p>
      </div>
    </div>
  </div>
  <div
    id="pager"
  >
    <h4>
      <span
        id="pager-span"
        style={
          Object {
            "display": "flex",
            "justify-content": "space-around",
          }
        }
      >
        <a
          href="/1"
          style={
            Object {
              "color": "#D3D3D3",
              "cursor": "default",
              "pointer-events": "none",
            }
          }
        >
          ←
          Back
        </a>
        <a
          href="/1"
          style={
            Object {
              "color": "#D3D3D3",
              "cursor": "default",
              "pointer-events": "none",
            }
          }
        >
          More 
          →
        </a>
      </span>
    </h4>
  </div>
</div>
`;
